packages/nestjs/src/cron-decorator.ts

```diff
@@ -0,0 +1,24 @@
+import * as Sentry from '@sentry/node';
+import type { MonitorConfig } from '@sentry/types';
+
+/**
+ * A decorator wrapping the native nest Cron decorator, sending check-ins to Sentry.
+ */
+export const SentryCron = (monitorSlug: string, monitorConfig?: MonitorConfig): MethodDecorator => {
+  return (target: unknown, propertyKey, descriptor: PropertyDescriptor) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const originalMethod = descriptor.value as (...args: any[]) => Promise<any>;
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    descriptor.value = async function (...args: any[]) {
+      return Sentry.withMonitor(
+        monitorSlug,
+        async () => {
+          return originalMethod.apply(this, args);
+        },
+        monitorConfig,
+      );
+    };
```

Comment:
h: Let's not modify the return value:

```suggestion
    descriptor.value = function (...args: any[]) {
      return Sentry.withMonitor(
        monitorSlug,
        () => return originalMethod.apply(this, args),
        monitorConfig,
      );
    };
```